Common/TableProducer/hmpConverter.cxx

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+// HMPID converter to new format
+// to be used with Run 2 converted data and older AO2Ds
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/AnalysisDataModel.h"
+
+using namespace o2;
+using namespace o2::framework;
+
+struct hmpConverter {
+  Produces<aod::HMPID_001> HMPID_001;
+
+  void process(aod::HMPID_000 const& hmpLegacy, aod::Tracks const&)
+  {
+    for (auto& hmpData : hmpLegacy) {
+
+      float phots[] = {0., 0., 0., 0., 0., 0., 0., 0., 0., 0.};
+
+      auto track = hmpData.track();
+      auto hmpidSignal = hmpData.hmpidSignal();
+      auto hmpidXTrack = -999.; // dummy
+      auto hmpidYTrack = -999.; // dummy
+      auto hmpidXMip = -999.;   // dummy
+      auto hmpidYMip = -999.;   // dummy
+      auto hmpidNPhotons = hmpData.hmpidNPhotons();
+      auto hmpidQMip = hmpData.hmpidQMip();
+      auto hmpidClusSize = -999; // dummy
+      auto hmpidMom = -999;      // dummy
+      auto hmpidPhotsCharge = phots;
+
+      HMPID_001(track,
```

Comment:
Thanks a lot! However, here I think you need the track *index*, not the track; you should probably be using .trackId() directly and in fact I do not think you need the tracks subscription at all for this converter. I guess as is now, with track() instead of trackId(), this converter will probably even crash... Not sure. Did you try to run it?

Reply:
so instead of track = hmpData.track(), you suggest to do trackid = hmpData.trackId()?
I just compiled, I have not run it.

Reply:
Yes exactly, and then when filling the table provide that trackId - that should be enough I think! 

And perhaps it would be good to try and run it. It won't do anything particularly useful because it will generate the table in version 001 that won't be used for now but it would be good to check that it does not crash :-D

Reply:
to run it I should have aod in the 000 format I guess, and then how the converter is used?

Reply:
Well actually I think format 000 is what we currently store everywhere so any existing AO2D should work fine (unless I am overlooking something)... So basically you can just run your converter right away - and if you remove the tracks table subscription I think you won't have any dependencies, even, so it should be straightforward to test!

Reply:
David the 000 format has never used. There was just an HMPID table in AnalysisDataModel.h as place holder (not implemented by me), when I started to implement the AO2D for the HMPID I modified that place holder table and then Jan raise the point about the backward compatibility, so I implemented versions 000 and 001, but at the moment AO2D for the HMPID doesn't exist!